stl/inc/ranges

```diff
@@ -4167,24 +4167,24 @@ namespace ranges {
 
             template <class _Rng>
             _NODISCARD static _CONSTEVAL _Choice_t<_St> _Choose() noexcept {
-                if constexpr (bidirectional_range<_Rng>) {

```

Comment:
I removed the `bidirectional_range<_Rng>` check because

1. [\[range.reverse.overview\]/2](http://eel.is/c++draft/range.reverse.overview#2) does not seem to have this
2. this breaks the following (contrived) code:

```c++
#include <ranges>
#include <iterator>
#include <istream>

using namespace std::ranges;
using namespace std::views;

struct weird : view_interface<weird> {
    int* begin();
    int* end();
    std::istreambuf_iterator<char> begin() const;
    std::istreambuf_iterator<char> end() const;
};

int main() {
    const auto v1 = weird{};
    auto v2 = reverse(v1); // rejected by MSVC STL, accepted by libstdc++
    auto v3 = reverse_view{v1}; // accepted by MSVC STL and libstdc++
}
```

Reply:
I must have been thinking this was `bidirectional_range<remove_cvref_t<_Rng>>`?  Even if that is correct, checking it first seems to make the first 3 cases more expensive and has no effect on the `reverse_view` case. I'll put this in the "I don't know what I was thinking, but I know I was wrong" bucket. (No change requested.)